src/main/kotlin/com/cognifide/gradle/aem/tooling/tail/Log.kt

```diff
@@ -0,0 +1,55 @@
+package com.cognifide.gradle.aem.tooling.tail
+
+import com.cognifide.gradle.aem.common.Formats
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+class Log(
+    val text: String,
+    val timestamp: LocalDateTime,
+    val level: String,
+    val source: String,
+    messageLines: List<String>
+) {
+
+    val checksum = Formats.calculateChecksum(text)
+    val message = messageLines.joinToString("\n")
+    val messageChecksum = Formats.calculateChecksum(message)
+
+    fun isError() = level == "ERROR"
+
+    fun isOlderThan(minutes: Long = 0, seconds: Long = 0) = LocalDateTime.now().minusMinutes(minutes).minusSeconds(seconds).isAfter(timestamp)
+
+    companion object {
+        private const val TIMESTAMP = """(?<timestamp>[0-9]{2}\.[0-9]{2}\.[0-9]{4}\s[0-9]{2}:[0-9]{2}:[0-9]{2}\.[0-9]{3})"""
+        private const val LEVEL = """\*(?<level>[A-Z]+)\*"""
+        private const val SOURCE = """(?<source>\[.*\])"""
+        private const val MESSAGE = """(?<message>.*)"""
+        private const val LOG_PATTERN = "$TIMESTAMP\\s$LEVEL\\s$SOURCE\\s$MESSAGE"
+        private const val DATE_TIME_FORMAT = "dd.MM.yyyy HH:mm:ss.SSS"
+        private val DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern(DATE_TIME_FORMAT)
+
+        fun create(logLines: List<String>): Log {
+            if (logLines.isEmpty() || logLines.first().isBlank()) throw TailException("Passed log entry is empty!")
+            val fullLog = logLines.joinToString("\n")
+            val result = matchLogLine(logLines.first())
+            when (result) {
+                null -> throw TailException("Passed text is not a log entry\nPattern:\n$LOG_PATTERN\nText:\n${logLines.first()}")
```

Comment:
should any edge case break the tailer by throwing exception? I am convinced that not, so that assure that such situation will be logged by warn/error (decide).